ArduinoCode/Driver-IO/dashboard/dashboard.ino

```diff
@@ -0,0 +1,148 @@
+// Author: Joshua Cheng
+// Date: November 3, 2020
+
+#include <mcp_can.h> // uses seeed-studio's CAN_BUS_Shield library
+#include <mcp_can_dfs.h>
+#include <SPI.h>
+
+const int spiCSPin = 10; // Base CAN pin
+MCP_CAN CAN(spiCSPin); // Set CS pin
+
+const unsigned char ON[] = {1}; // Message for turning inverter on
+const unsigned char OFF[] = {0}; // Message for turning inverter off
+const unsigned char FORWARD[] = {1}; // Message for setting inverter to forward
+const unsigned char REVERSE[] = {0}; // Message for setting inverter to reverse
+
+const int SHUTDOWN_DETECTION_PIN = 2; // Pin for detecting shutdown system voltage.
+const int BUTTON_PIN = 3; // Pin for on/off button.
+const int GEAR_SELECTOR_PIN = 4; // Pin for forward/reverse selector
+const int IMD_INPUT_PIN = 5; // Pin for detecting IMD status
+const int BMS_INPUT_PIN = 6; // Pin for detecting BMS status
+
+const int IMD_INDICATOR_PIN = 8; // Pin for IMD indicator
+const int BMS_INDICATOR_PIN = 9; // Pin for BMS indicator
+const int BUTTON_LED_PIN = 11; // Pin for integrated LED on the button.
+const int SPEAKER_PIN = 10; // Pin for speaker or relay for speaker (must be pwm pin).
+
+bool shutdownVoltage = false;
+unsigned long speakerCooldown = 0; // Use unsigned long to match millis() function. Will overflow in around 50 days.
+unsigned long buttonCooldown = 0; // Use unsigned long to match millis() function. Will overflow in around 50 days.
```

Comment:
Will these end up overflowing? Why?

Reply:
They would never overflow. Our battery will die within a few hours, so even a 1 day overflow time would be acceptable. @jamescd18 this code will be entirely restarted on power cycle of the car